### ISSUE TITLE
Error running 'custodian report' in c7n version 0.8.33.0

### ISSUE THREAD

**Issue by tzd-sh** [cloud-custodian/cloud-custodian#3301]
The current version of c7n 0.8.33.0 appears to have broken the 'custodian report' functionality.

Below is my troubleshoot steps, showing that it works when rolled back to 0.8.32.0

```
$ pip list|grep c7n
c7n                0.8.33.0
c7n-org            0.5.0


$ cat list-ec2-instances.yaml
policies:
  - name: list-ec2-instances
    resource: ec2


$ custodian run -s ~/output -r us-east-1 list-ec2-instances.yaml
2019-01-03 22:20:42,075: custodian.policy:INFO policy: list-ec2-instances resource:ec2 region:us-east-1 count:1 time:0.37


$ custodian report -s ~/output -r us-east-1 list-ec2-instances.yaml
Traceback (most recent call last):
  File "/home/ec2-user/custodian/bin/custodian", line 11, in <module>
    sys.exit(main())
  File "/home/ec2-user/custodian/local/lib/python2.7/site-packages/c7n/cli.py", line 368, in main
    command(config)
  File "/home/ec2-user/custodian/local/lib/python2.7/site-packages/c7n/commands.py", line 105, in _load_policies
    p_options = provider.initialize(options)
  File "/home/ec2-user/custodian/local/lib/python2.7/site-packages/c7n/resources/aws.py", line 399, in initialize
    XrayTracer.initialize()
  File "/home/ec2-user/custodian/local/lib/python2.7/site-packages/c7n/resources/aws.py", line 222, in initialize
    xray_recorder.configure(
NameError: global name 'xray_recorder' is not defined


$ cat rollback.txt
c7n==0.8.32.0


$ pip install --upgrade --force-reinstall -r rollback.txt


$ pip list|grep c7n
c7n                0.8.32.0
c7n-org            0.5.0

$ custodian report -s ~/output -r us-east-1 list-ec2-instances.yaml
CustodianDate,InstanceId,tag:Name,InstanceType,LaunchTime,VpcId,PrivateIpAddress
2019-01-03 22:20:42.074099,i-028454455c3bb6e1c,Cloud Custodian Host,t3.micro,2019-01-03T21:23:51+00:00,vpc-b869c1c0,172.31.37.231

```

**Comment by kapilt** [comment]
ugh, thanks for the bug report. a temporary workaround for 0.8.33.0 is to install `aws_xray_sdk` python package.

i think part of the issue with this getting into a release is that the ci infrastructure, always ends up running with the xray sdk installed, but its not intended to be a hard runtime dependency, just an optional, but there isn't a clean way to verify that. It looks like the actual pr tracks back to some cli lazy load optimizations for latency gains.

**Comment by kapilt** [comment]
released 0.8.33.1 with the fix
